### DISCUSSION TITLE
Container Overwrote Previous World

### DISCUSSION THREAD

**Discussion by AtroposOrbis** [mbround18/valheim-docker#101] [Help!] [answered]
When I performed `docker-compose pull valheim` -> `docker-compose up -d` the server overwrote the database for our world, effectively resetting the world back to the way it was prior to anyone ever having stepped foot on it. Is their anyway that the server saved a .old version of the world somewhere?? We were at day 104 T_T

**Comment by mbround18** [comment, marked as answer]
> When I performed `docker-compose pull valheim` -> `docker-compose up -d` the server overwrote the database for our world, effectively resetting the world back to the way it was prior to anyone ever having stepped foot on it. Is their anyway that the server saved a .old version of the world somewhere?? We were at day 104 T_T

`docker-compose pull valheim` should not have reset your world? It sounds like you got hit by the current world eater bug :/ sorry man.... I am including a backup feature because of this prevalent bug thats going on currently.

**Reply by AtroposOrbis** [reply]
Will the backups be baked directly into the volumes as well? Or will it be additional configuration? I don't want my peeps to go through this again....

**Reply by mbround18** [reply]
I have a full write up in the readme <3 there are many fail safes I am including such as backups on update, shutdown, and on an interval. 

![image](https://user-images.githubusercontent.com/12646562/108001810-654f1a00-6fa2-11eb-82f4-a9e2393b9298.png)

**Reply by AtroposOrbis** [reply]
@mbround18 You're a saint and a scholar good sir. Thank you for all the hard work and I promise to be sparing in my creation of additional subjects. <3

**Comment by mbround18** [comment]
@AtroposOrbis auto update feature is live! I am now running it on my server and it looks like its running fine :) 

See the everything and the kitchen sink section in the read me for an example on how to use all the variables. You will have to add an additional volume mount for it to work.

**Reply by ChristopherHMelgar** [reply]
Hi yesterday i did dokcer pull to get the latest version, my world came back many days to go back (the day I uploaded my local world to the server) I see backups files in `./valheim/backups` but I don't know how to restore them 
![image](https://user-images.githubusercontent.com/33786947/112349143-3fa2e800-8ca7-11eb-81be-e51b1942a160.png)
why? the size of my files is 0
![image](https://user-images.githubusercontent.com/33786947/112349395-72e57700-8ca7-11eb-8d64-9857df006502.png)
this is my docker compose file
![image](https://user-images.githubusercontent.com/33786947/112349545-94466300-8ca7-11eb-976b-4fa2b2b832ba.png)
and this the images, i guess the one with tag `none` is the one that worked fine before the pull
Do you think there is a way that I can recover my world?

**Reply by mbround18** [reply]
That screams permissions issues, did you change permissions or have different permissions in any way? Check your host file and see if the world files exist. 

The container itself does NOT write to world files in any way shape or form and it does NOT manage world files or change them. All the container does is read the files from the world directory and add them to a zip folder.

**Reply by ChristopherHMelgar** [reply]
I always had the same permissions I run `docker-compose up` with `sudo`, is this bad?
there are the files of my world
![image](https://user-images.githubusercontent.com/33786947/112356649-8dbaea00-8cad-11eb-8888-4164b3505b39.png)
change the permissions of the last backup where my world was fine to try, bat is the same (the initial world)
![image](https://user-images.githubusercontent.com/33786947/112356868-c4910000-8cad-11eb-9fa0-85b51708ba48.png)
how correct the permissions issues? when shutdow the server the backup workfine see the size, I don't know why the previous auto backups only have 0 size.
what I don't understand is how the world files are updated

**Reply by CosmicHorrorDev** [reply]
Just curious. How much disk space do you have left?

Regardless you can restore a backup by extracting the contents and replacing your saves mount with it (it's just a compressed tarball so it should look familiar when you extract it)

I'm not sure why the auto-backups would be failing

**Reply by ChristopherHMelgar** [reply]
![image](https://user-images.githubusercontent.com/33786947/112382737-fe243400-8cca-11eb-8609-64d72399b583.png)
i have 47G space left

now the backups works fine, but the previus backups has 0 size and it has nothing to extract.
![image](https://user-images.githubusercontent.com/33786947/112383271-ab974780-8ccb-11eb-8dab-73dcae5b82e3.png)
I think I lost my world progress, I will be monitoring if the backups are generated correctly
